### PR TITLE
[#531719829]Add Location Update endpoint

### PR DESCRIPTION
**Pull request by ShabiniRajadas**
### Description

Adding update endpoint to Location. Allowing changes to be made to the max_capacity attribute.

Added timezone and timezone_valid attribute. Because many locations have an invalid timezone, so if a location has an invalid timezone, we will set the value of timezone_valid to be false and pop up notification in the mobile, asking them to update to a valid timezone. The frontend will also be updated with a valid set of timezones. 

### Changelog
- Added model - locationUpdateParams
- Added max_capacity, timezone, and timezone_valid to location model
- Added update endpoint for PUT /locations/{location_id}

### CODE REVIEW

**Comment by tractionguest-integrations** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  Remember to use the `Squash and Merge` button to merge!
    Always remember, your final commit message needs to start with a wrike ticket like, `[#1234] Your commit message`.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  Associated Wrike Tickets:
<a href='https://www.wrike.com/open.htm?id=531719829' target="_blank">#531719829</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  <a href="https://stoplight.io/p/docs/gh/tractionguest/guest-api/openapi.yml?srn=gh/tractionguest/guest-api/openapi.yml&group=location-update" target=_blank>View docs for this page</a>
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against de831fa48cfdd0acb43c9be510f705963c710420
</p>

File: models/Location.v1.yaml [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
-  - id
-  - name
 type: object
+x-examples:
+  default:
+    id: 1
+    name: Default
+    max_capacity: 30
+    timezone_valid: true
+    timezone: 'Central Time (US & Canada)'
```
Comment (stevenallen05) [inline models/Location.v1.yaml:10]: The timezones should be following the TZ database maintained by the IANA (eg: `America/Vancouver`, `Europe/London`, etc)

Comment (ShabiniRajadas) [reply models/Location.v1.yaml:10]: 👍

File: openapi.yml [hunk 2]
```diff
@@ -3345,6 +3345,73 @@ paths:
...
+            application/json:
+              schema:
+                $ref: ./models/Location.v1.yaml
+              examples:
+                default:
+                  value:
+                    id: 1
+                    name: Default
+                    max_capacity: 30
+                    timezone: 'Central Time (US & Canada)'
```
Comment (clrobinson-traction) [inline openapi.yml:3372]: I think Steven's timezone comment applies here too.

File: openapi.yml [hunk 3]
```diff
@@ -3345,6 +3345,73 @@ paths:
...
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: ./models/LocationUpdateParams.v1.yaml
+            examples:
+              default:
+                value:
+                  max_capacity: 30
+                  timezone: 'Central Time (US & Canada)'
```
Comment (clrobinson-traction) [inline openapi.yml:3413]: Again, previous timezone comment.

File: openapi.yml (hunk 2 above)
Comment (stevenallen05) [inline openapi.yml:3372]: This example should be updated to use the IANA TZ designations, too.

File: openapi.yml (hunk 3 above)
Comment (stevenallen05) [inline openapi.yml:3413]: This example should be updated to use the IANA TZ designations, too.

**Comment by ShabiniRajadas** [comment]
The location update feature is descoped for the current quarter. So closing this PR.

**Pull request closed without merging**